openadapt/window/_windows.py

```diff
@@ -1,30 +1,155 @@
+    data = get_element_properties(active_window)
     state = {
-        "title": title,
-        "left": left,
-        "top": top,
-        "width": width,
-        "height": height,
-        # TODO: get window state; see:
-        # https://github.com/MLDSAI/OpenAdapt/issues/75#issuecomment-1536762953
-        "meta": None,
-        "data": None,
-        "window_id": None,
+        "title": meta["texts"][0],
+        "left": meta["rectangle"].left,
+        "top": meta["rectangle"].top,
+        "width": meta["rectangle"].width(),
+        "height": meta["rectangle"].height(),
+        "meta": {**meta, "rectangle": rectangle_dict},
+        "data": data,
+        "window_id": meta["control_id"],
     }
+    try:
+        pickle.dumps(state)
+    except Exception as exc:
+        logger.warning(f"{exc=}")
+        state.pop("data")
     return state
 
 
-def get_element_at_position(x, y):
-    # TODO
-    return None
+def get_active_window_meta(active_window) -> dict:
+    """
+    Get the meta information of the active window.
+
+    Args:
+        active_window: The active window object.
+
+    Returns:
+        dict: A dictionary containing the meta information of the active window.
+    """
+    if not active_window:
+        logger.warning(f"{active_window=}")
+        return None
+    result = active_window.get_properties()
+    return result
+
+
+def get_active_element_state(x: int, y: int):
+    """
+    Get the state of the active element at the given coordinates.
+
+    Args:
+        x (int): The x-coordinate.
+        y (int): The y-coordinate.
+
+    Returns:
+        dict: A dictionary containing the properties of the active element.
+    """
+    active_window = get_active_window()
+    active_element = active_window.from_point(x, y)
+    properties = active_element.get_properties()
+    properties["rectangle"] = serialize_rect(properties["rectangle"])
+    return properties
+
+
+def get_active_window(depth=10, max_width=10, filename=None) -> Desktop:
+    """
+    Get the active window object.
+
+    Returns:
+        Desktop: The active window object.
+    """
+    app = pywinauto.application.Application(backend="uia").connect(active_only=True)
+    window = app.active()
+    return window
+
+
+def get_element_properties(element):
+    """
+    Recursively retrieves the properties of each element and its children.
+
+    Args:
+        element: An instance of a custom element class that has the `.get_properties()` and `.children()` methods.
+
+    Returns:
+        A nested dictionary containing the properties of each element and its children.
+        The dictionary includes a "children" key for each element, which holds the properties of its children.
+
+    Example:
+        element = Element()
+        properties = get_element_properties(element)
+        print(properties)
+        # Output: {'prop1': 'value1', 'prop2': 'value2', 'children': [{'prop1': 'child_value1', 'prop2': 'child_value2', 'children': []}]}
```

Comment:
Can you please split this up (and other lines in this docstring) on multiple lines such that no line is greater than 80 characters?

Reply:
Done @abrichr